lib/chef/provider/package/cab.rb

```diff
@@ -0,0 +1,147 @@
+#
+# Author:: Vasundhara Jagdale (<vasundhara.jagdale@msystechnologies.com>)
+# Copyright:: Copyright 2015-2016, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/package"
+require "chef/resource/cab_package"
+require "chef/mixin/powershell_out"
+
+class Chef
+  class Provider
+    class Package
+      class Cab < Chef::Provider::Package
+        include Chef::Mixin::PowershellOut
+
+        provides :cab_package, os: "windows"
+
+        def load_current_resource
+          @current_resource = Chef::Resource::CabPackage.new(new_resource.name)
+          current_resource.source(new_resource.source)
+          new_resource.version(package_version)
+          current_resource.version(installed_version)
+          current_resource
+        end
+
+        def install_package(name, version)
+          dism_command("/Add-Package /PackagePath:\"#{@new_resource.source}\"")
+        end
+
+        def remove_package(name, version)
+          dism_command("/Remove-Package /PackagePath:\"#{@new_resource.source}\"")
+        end
+
+        def dism_command(command)
+          powershell_out("dism.exe /Online #{command} /NoRestart", { :timeout => @new_resource.timeout })
```

Comment:
I would use `Mixlib:ShellOut` over `powershell_out` since we are not leveraging powershell commands and powershell_out just adds command layers on top of shellout.